File: frameworks/domain.core/src/main/java/com/example/frameworks/domain/core/GlobalIdGenerator.java
package com.example.frameworks.domain.core;

import lombok.*;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class GlobalIdGenerator {
    private static IdGenerator idGenerator = new UuidBasedIdGenerator();

    public static String nextId() {
        return idGenerator.next();
    }

    public static String nextReadableId() {
        return idGenerator.nextReadable();
    }

    static void reset(IdGenerator idGenerator) {
        GlobalIdGenerator.idGenerator = idGenerator;
    }
}

File: frameworks/domain.core/src/main/kotlin/com/qomolangma/frameworks/domain/core/GlobalIdGenerator.kt
package com.qomolangma.frameworks.domain.core

import lombok.AccessLevel
import lombok.NoArgsConstructor

@NoArgsConstructor(access = AccessLevel.PRIVATE)
object GlobalIdGenerator {
    private var idGenerator: Id = UuidBasedId()

    @JvmStatic
    fun nextId(): String {
        return idGenerator.next()
    }

    @JvmStatic
    fun nextReadableId(): String {
        return idGenerator.nextReadable()
    }

    @JvmStatic
    fun reset(idGenerator: Id) {
        GlobalIdGenerator.idGenerator = idGenerator
    }
}
